sycl/test-e2e/VirtualMem/test-plan.md

```diff
@@ -0,0 +1,140 @@
+
+#### Reserving virtual address ranges
+
+A check should be performed that exception with `feature_not_supported` error
+code is thrown when `context` passed to it contains multiple devices and not
+every of them have `ext_oneapi_virtual_mem` aspect.
+
+A check should be performed that we can successfully perform and immediately
+release a valid reservation.
+
+**TODO**: the spec currently says that passing any invalid arguments to those
+APIs results in UB and therefore there is nothing to test. This is probably an
+incorrect approach of the spec, because we should be able to perform some checks
+and inform user about an error. If/when the spec is updated, this test plan
+should be updated as well to cover those new scenarios.
+
+#### Physical memory
+
+A check should be performed that methods `get_context()`, `get_device()` and
+`size()` return correct values (i.e. ones which were passed to `physical_mem`
+constructor).
+
+A check should be performed that a value returned from a valid call to `map()`
+is the same as `reinterpret_cast<void *>(ptr)`.
+
+**TODO**: the spec currently doesn't say what happens if `map` arguments are
+invalid (misaligned `ptr`, or `offset + numBytes > size()`) and once that is
+clarified the test plan should be updated to cover those scenarios as well.
+
+A check should be performed that we can change access mode of a virtual memory
+range and immediately see it changed.
+
+A check should be performed that we can successfully map and immediately unmap
+a virtual memory range.
+
+### End-to-end tests
+
+Tests in this category perform some meaningful actions with the extension to
+see that the extension works in a scenarios which mimic real-life usage of the
+extension.
+
+#### Basic access from a kernel
+
+A test should allocate some amount of a physical memory and then create a
+virtual memory range mapped to that physical memory with access mode read-write.
+Test should submit two kernels:
+- the first kernel fills virtual memory with some data
+- the second kernel reads from that virtual memory and copies it content into
+  an accessor
+
+Before the second kernel is submitted, we should change access mode of the
+virtual memory range to be read-only.
+
+The content of the accessor is then checked on host to ensure that data can be
+correctly both written and read from virtual memory.
+
+#### Ability to use virtual memory as USM memory
+
+This test case is intended to check that a pointer produced by a virtual memory
+range mapping can indeed be used in various APIs accepting a USM pointer.
+
+A series of checks should be performed:
+- that we can copy _from_ virtual memory _to_ a buffer via accessor
+- that we can copy _from_ virtual memory _to_ a USM allocation
+- that we can copy _from_ a USM allocation _to_ virtual memory
+- that we can copy _from_ a buffer via accessor _to_ virtual memory
+- that we can use `memset` on virtual memory
+- that we can use `fill` on virtual memory
+
```

Comment:
I think it should be an explicit exception in the spec about `sycl::free` because the core spec says the pointer must have been allocated using one of the USM allocation functions.

Reply:
I agree, `free` should be excluded by the extension. Maybe add another todo to mention this open?